pages/profile.js

```diff
@@ -14,13 +14,45 @@ import { blueGrey } from '@mui/material/colors';
 import ProfilePost from '../components/Profile/ProfilePost';
 import ProfileFollow from '../components/Profile/ProfileFollow';
 import UserAdd from '../icons/user-add';
-import { follow } from '../services/followServices';
+import { followUser, getFollowedStatus } from '../services/Follow';
+import { getCurrentUser } from '../services/Users';
 import hotToast from '../utils/hotToast';
 import newNotification from '../utils/newNotification';
 
-const Profile = () => {
+const Profile = ({ username }) => {
   const [currentTab, setCurrentTab] = useState('posts');
   const [followedStatus, setFollowedStatus] = useState('not_followed');
+  const [currentName, setCurrentName] = useState('');
+  // const [role, setRole] = useState('null');
+  const [currentRole, setCurrentRole] = useState('');
+
+  // Get current user details
+  useEffect(() => {
+    const getUser = async () => {
+      const { data } = await getCurrentUser();
+      console.log(data);
+      setCurrentName(data.username);
+      setCurrentRole(data.role.roleName);
+      if (!username) {
+        setFollowedStatus('current_user');
+      }
+    };
+    getUser();
+  }, [username]);
+
+  // Check follow status
+  useEffect(() => {
+    const checkStatus = async (name) => {
+      const { data } = await getFollowedStatus(name);
+      if (data === true) {
```

Comment:
what is the data type? must this data be `true`, or you mean the `data` exists or not

Reply:
The data type is Boolean. Should it be "if (data) {"?